tools/src/main/scala/scala/scalanative/build/Build.scala

```diff
@@ -118,7 +118,13 @@ object Build {
           backend
             .codegen()
             .flatMap(backend.compile)
-            .map(backend.link)
+            .map { p =>
+              val linked = backend.link(p)
+              if (Platform.isMac && config.compilerConfig.debugMetadata) {
+                backend.dsymutil(linked)
+              }
+              linked
+            }
```

Comment:
Maybe let's create an additional method `backend.postprocess` instead which would call the dsymutil and possibly other non-crucial post processing operations.